### DISCUSSION TITLE
Robot alternative

### DISCUSSION THREAD

**Discussion by hecodeit** [cmubig/Frida#6] [Q&A] [answered]
Really impressive project, like to learn  the project and workflow.  

The problem is don't get the same robotic arm. My question is other robotic arm work with this project? Already have desktop robotic arm but low repeatability, could cause the results?

**Comment by pschaldenbrand** [comment]
Thanks @hecodeit 

Low repeatability could be an issue. Do you know how far off the arm is? If it's a matter of a few millimeters, I think the FRIDA codebase will work fine with some small degradation in quality. But if it is off by centimeters it would probably not work well.  You can take a look at src/robot.py to see how you can create a new robot class for controlling your robot. Thanks for your interest in the project!

**Reply by hecodeit** [reply]
+-0.5 mm, +- 1.0 mm robotic arm?

**Reply by pschaldenbrand** [reply]
Yeah that should be totally fine. Is it a custom robot? If not, I'd really appreciate if you could submit a pull request to add the robot code whenever you get it working.

**Reply by hecodeit** [reply]
DIY HBot with OpenBuilds years ago. For now, robotic painting purpose, struggle which robotic arm should get, maybe cobot MG400, Nova 2, or UFactory Lite 6? I like UFactory Lite 6, cheaper with 6 axis...

**Reply by pschaldenbrand** [reply, marked as answer]
I use the UFactory Lite 6 for Sharpie marker drawings with FRIDA. I think it's great. The problem is it doesn't have a ton of reach so the drawings are small. I haven't tried using it for paintings, but I think it could be possible. You would probably just have to use small canvases and less colors.
